### PR TITLE
page docs

### LINKED ISSUE #236: Page Class XML Documentation

### PR DESCRIPTION
**Pull request by Meir017**
closes #236

### CODE REVIEW

File: lib/PuppeteerSharp/Page.cs [hunk 1]
```diff
@@ -63,28 +76,97 @@ private Page(Session client, Target target, FrameTree frameTree, bool ignoreHTTP
...
+        /// <summary>
+        /// Emitted when a <see cref="Response"/> is received.
+        /// </summary>
         public event EventHandler<ResponseCreatedEventArgs> ResponseCreated;
+
+        /// <summary>
+        /// Emitted when a page issues a request. The <see cref="Request"/> object is read-only.
+        /// In order to intercept and mutate requests, see <see cref="SetRequestInterceptionAsync(bool)"/>
+        /// </summary>
         public event EventHandler<RequestEventArgs> RequestCreated;
```
Comment (Meir017) [inline lib/PuppeteerSharp/Page.cs]: I think that this should be rename to `Request` to match puppeteer

File: lib/PuppeteerSharp/Page.cs [hunk 2]
```diff
@@ -63,28 +76,97 @@ private Page(Session client, Target target, FrameTree frameTree, bool ignoreHTTP
...
+
+        /// <summary>
+        /// Emitted when a frame is navigated to a new url.
+        /// </summary>
         public event EventHandler<FrameEventArgs> FrameNavigated;
 
+        /// <summary>
+        /// Emitted when a <see cref="Response"/> is received.
+        /// </summary>
         public event EventHandler<ResponseCreatedEventArgs> ResponseCreated;
```
Comment (Meir017) [inline lib/PuppeteerSharp/Page.cs]: I think that this should be rename to `Response` to match puppeteer

File: lib/PuppeteerSharp/Page.cs [hunk 3]
```diff
@@ -63,28 +76,97 @@ private Page(Session client, Target target, FrameTree frameTree, bool ignoreHTTP
...
+
+        /// <summary>
+        /// Emitted when the page crashes
+        /// </summary>
         public event EventHandler<ErrorEventArgs> Error;
+
+        /// <summary>
+        /// Emitted when the JavaScript code makes a call to <c>console.timeStamp</c>. For the list of metrics see <see cref="Page.MetricsAsync"/>.
+        /// </summary>
         public event EventHandler<MetricEventArgs> MetricsReceived;
```
Comment (Meir017) [inline lib/PuppeteerSharp/Page.cs]: I think that this should be renamed to `Metrics` to match puppeteer

File: lib/PuppeteerSharp/Page.cs (hunk 1 above)
Comment (kblok) [reply lib/PuppeteerSharp/Page.cs]: Though I don't like the idea, I think we had this conversation about being strong matching puppeteer. Let's do it

File: lib/PuppeteerSharp/Page.cs (hunk 2 above)
Comment (kblok) [reply lib/PuppeteerSharp/Page.cs]: Same

File: lib/PuppeteerSharp/Page.cs (hunk 3 above)
Comment (kblok) [reply lib/PuppeteerSharp/Page.cs]: Same

File: lib/PuppeteerSharp/Page.cs [hunk 4]
```diff
@@ -55,54 +68,159 @@ private Page(Session client, Target target, FrameTree frameTree, bool ignoreHTTP
...
 
             Client.MessageReceived += client_MessageReceived;
         }
 
+        internal Session Client { get; }
+
         #region Public Properties
+
+        /// <summary>
+        /// Emitted when the JavaScript <c>load</c> <see cref="https://developer.mozilla.org/en-US/docs/Web/Events/load"/> event is dispatched.
```
Comment (kblok) [inline lib/PuppeteerSharp/Page.cs]: In C# I think the proper word should be `raised` rather than `emitted`

File: lib/PuppeteerSharp/Page.cs [hunk 5]
```diff
@@ -126,6 +244,13 @@ private Page(Session client, Target target, FrameTree frameTree, bool ignoreHTTP
 
         #region Public Methods
 
+        /// <summary>
+        /// Returns metrics
+        /// </summary>
+        /// <returns>metrics</returns>
```
Comment (kblok) [inline lib/PuppeteerSharp/Page.cs]: I think it would be good to follow the wording "Task which resolves to" a list of metrics" (across the class)

**Pull request merged**